flight/mcu/lib/teensyShared/rocketFSM.cpp

```diff
@@ -30,32 +30,38 @@
 #include "pins.h"
 #include "thresholds.h"
 
-rocketFSM::rocketFSM(pointers *ptr) { pointer_struct = ptr; }
+rocketFSM::rocketFSM(pointers *ptr) {
+    pointer_struct = ptr;
+    // get the linear accelration from the lowgimu
+    float linear_acceleration = pointer_struct->sensorDataPointer->highG_data.hg_az;
+    // storing pointer_struct variables locally for use
+    FSM_State rocket_state = pointer_struct->sensorDataPointer->rocketState_data.rocketState;
+    uint32_t launch_time = pointer_struct->rocketTimers.launch_time;
+    uint32_t burn_timer = pointer_struct->rocketTimers.burn_timer;
+    uint32_t burnout_time = pointer_struct->rocketTimers.burnout_time;
+    uint32_t coast_timer = pointer_struct->rocketTimers.coast_timer;
+}
 
 void rocketFSM::tickFSM() {
     // Lock mutexes for data used in switch
     chMtxLock(&pointer_struct->dataloggerTHDVarsPointer.dataMutex_highG);
 
-    // get the linear accelration from the lowgimu
-    float linear_acceleration =
-        pointer_struct->sensorDataPointer->highG_data.hg_az;
-
```

Comment:
We need to update linear acceleration for the fsm logic

Reply:
We should probably store the pointer to hg_az as a member variable. If we do that we can either just reference that in the rest of the code or store a local variable to release the mutex. I'm not sure if Chibios is running round-robin or if it somehow switches in the middle of threads, but storing it as a local variable within the tick function is probably just as efficient either way.